src/3d/qgstiledscenechunkloader_p.cpp

```diff
@@ -33,16 +36,17 @@ size_t qHash( const QgsChunkNodeId &n )
   return n.uniqueId;
 }
 
-static bool hasLargeBounds( const QgsTiledSceneTile &t )
+static bool hasLargeBounds( const QgsTiledSceneTile &t, const QgsCoordinateTransform &boundsTransform )
 {
   if ( t.geometricError() > 1e6 )
     return true;
 
   if ( t.boundingVolume().box().isNull() )
     return true;
 
-  const QgsVector3D size = t.boundingVolume().box().size();
-  return size.x() > 1e5 || size.y() > 1e5 || size.z() > 1e5;
+  Q_ASSERT( boundsTransform.destinationCrs().mapUnits() == Qgis::DistanceUnit::Meters );
+  auto bounds = t.boundingVolume().bounds( boundsTransform );
```

Comment:
QGIS coding convention is to avoid auto for trivial types (ie only use it for iterators)